docs/hugo/content/architecture-decision-records/ADR-2022-02-Adoption-Policy.md

```diff
@@ -0,0 +1,127 @@
+`kubectl delete` always deletes the backing resource no matter what.
+
+[GCP](https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources#creating_a_resource) does it this way:
+> When you create a resource, Config Connector creates the resource if it doesn't exist. If a Google Cloud resource 
+> already exists with the same name, then Config Connector acquires the resource and manages it.
+
+GCP also [supports](https://cloud.google.com/config-connector/docs/how-to/managing-deleting-resources) a 
+`reconcile-policy`-like annotation, so doing it like this would basically be doing it exactly like GCP.
+
+Kuberentes has a similar concept for `StorageClasses`/`PersistentVolumes` called 
+[reclaimPolicy](https://kubernetes.io/docs/concepts/storage/storage-classes/#reclaim-policy), whose default
+seems to be `Delete`.
+
+#### Option 2: The default `reconcile-policy` for adopted resource is `abandon-on-delete`.
+
+ASO detects that a resource previously existed and sets `reconcile-policy` automatically to `abandon-on-delete`.
+A new option in the ASO config `adoption-policy` would control this default so users could opt into the Option 1 style
+discussed above if they wanted (but it's not the default behavior).
+
+**Advantages:**
+- More flexibility. Users can choose which behavior they want when deploying ASO. 
+- Defaults to a more conservative `reconcile-policy` to protect them from accidentally deleting things they may not
+  have wanted to delete.
+
+**Disadvantages:**
+- Slightly higher complexity. Users will be exposed to `reconcile-policy` as set by the operator (when a resource is adopted)
+  so need to understand it.
+- Bias towards safety may cause resource user thinks was deleted to continue existing (and billing) in Azure. Not a data
+  loss though.
+- Implementation complexity: We would have to issue some extra requests to Azure to detect if the resource exists before `PUT`-ing it. 
+  
+  Since `reconcile-policy` as an annotation isn't managed solely by the user or the operator, deciding if
+  we should trigger reconciliation events on its change is tricky (we don't want our defaulting of it to trigger an event). 
+  It's doable though.
+
+**Thoughts/comments:**
+
+There's some discussion of a similar feature over in [ACK](https://github.com/aws-controllers-k8s/community/pull/1148#discussion_r794605821).
+Jay Pipes had an interesting comment:
+> In the case of ACK and the user experience of resource deletion, I posit that the behaviour of our interface has 
+> no surprises and is actually hard to misuse. It doesn't have surprises because our interface to delete a resource 
+> is to call kubectl delete on that resource. This is as non-surprising as it gets. It is hard to misuse because 
+> calling kubectl delete is an explicit action the user has to take and it's not a confusing verb or semantic. 
+> "delete" means to remove the resource.
```

Comment:
fully agree with this.

Reply:
I fully agree that there's no ambiguity that `kubectl delete` means remove the resource _from the cluster_.

That said, I do think that different users will have different expectations about whether deletion from the cluster necessarily requires deletion from Azure as a natural consequence.